server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -538,6 +496,21 @@ private static void softDeleteSecretSeries(DSLContext dslContext, SecretsRecord
         .where(SECRETS.ID.eq(record.getId()))
         .execute();
 
+    List<Field<?>> fieldsToCopy = Arrays.stream(DELETED_ACCESSGRANTS.fields())
+        .filter(field -> field.getName() != DELETED_ACCESSGRANTS.ID.getName())
```

Comment:
```suggestion
   .filter(field -> !field.getName().equals(DELETED_ACCESSGRANTS.ID.getName()))
```
nit: suggested from ide